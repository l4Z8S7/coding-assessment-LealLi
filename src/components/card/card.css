.card {
  border: 2px solid #260ba9;
  border-radius: 8px;
  width: 80%;
  min-width: 400px;
  min-height: 200px;
  margin: auto;
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: stretch;
  box-shadow: 9px 12px #595b61; 
}

.card .arrow {
  min-width: 50px;
  font-size: 36px;
  background-color: #66a1f4;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.card .arrow-left {
  border-radius: 8px 0 0 8px;
}

.card .arrow-right {
  border-radius: 0 8px 8px 0;
}

.card .question {
  width: 100%;
}

.card .question .title {
  font-size: 24px;
  font-weight: bold;
  padding: 10px;
  border-bottom: 1px double #0b0238;
}

.card .question .options {
  height: min-content;
  margin: 10px;
  padding: 0;
  list-style-type: none;
  display: flex;
  flex-direction: column;
  justify-content: flex-start; 
}

.card .question .options li {
  margin-bottom: 5px;
  height: min-content;
  border: 1px solid #29365a;
  border-radius: 5px;
  cursor: pointer;
}

.card .question .options li.selected {
  margin-bottom: 5px;
  height: min-content;
  background-color: #b2c5ee;
  border: 1px solid #29365a;
  border-radius: 5px;
}

.card .question .options li span {
  font-size: 16px;
  padding: 5px;
  display: inline-block;
}