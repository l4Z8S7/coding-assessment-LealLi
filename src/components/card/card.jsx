import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './card.css';
import * as actions from '../../actions';

const Card =  (props) => {
    const options = Object.keys(props.question.options);
    return (
        <div className="card">
            <div 
                className="arrow arrow-left" 
                onClick={() => props.prevQuestion()}
                onKeyPress={(event) => event.key === "Enter" ? props.prevQuestion() : void(0)}  
                style={props.index === 0 ? { backgroundColor: "#848591", pointerEvents: "none" } : null}
                tabIndex="1"
            ><i className="fa fa-angle-left"></i></div>
            <div className="question">
                <div className="title">
                { props.question.title }
                </div>
                <ul className="options">
                {
                    options.map(option => <li 
                        key={option} 
                        onClick={() => props.selectAnswer(props.index, option)}
                        onKeyPress={(event) => event.key === "Enter" ? props.selectAnswer(props.index, option) : void(0)}
                        className={option === props.answer ? "selected" : ""} 
                        tabIndex="1"
                    ><span>{option}</span></li>)
                }
                </ul>
            </div>
            <div 
                className="arrow arrow-right" 
                onClick={() => props.nextQuestion()}
                onKeyPress={(event) => event.key === "Enter" ? props.nextQuestion() : void(0)} 
                tabIndex="1"
            ><i className="fa fa-angle-right"></i></div>
        </div>
    );
}

function mapStateToProps({ questions, index, answers }) {
    return { question: questions[index], answer: answers[index], index };
}

Card.propTypes = {
    question: PropTypes.object.isRequired,
    answer: PropTypes.string,
    index: PropTypes.number.isRequired,
    prevQuestion: PropTypes.func.isRequired,
    nextQuestion: PropTypes.func.isRequired,
    selectAnswer: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, actions)(Card);