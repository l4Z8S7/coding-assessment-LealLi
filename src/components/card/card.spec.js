import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { Card } from './card';

Enzyme.configure({ adapter: new Adapter });

const props = {
  question: { title: 'Q1', options: { 'Opt1': 1, 'Opt2': 2} },
  answer: '',
  index: 0,
  prevQuestion: () => {},
  nextQuestion: () => {},
  selectAnswer: () => {},
}

describe('Card', () => {
  it('should render component matching the snapshot', () => {
    expect(toJson(shallow(<Card {...props} />))).toMatchSnapshot();
  });

  it('should change class name for selected answer', () => {
    expect(shallow(<Card {...props} answer={'Opt1'} />).find('li').first().hasClass('selected')).toBe(true);
  })
});