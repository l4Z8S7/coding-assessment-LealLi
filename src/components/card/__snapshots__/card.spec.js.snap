// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render component matching the snapshot 1`] = `
<div
  className="card"
>
  <div
    className="arrow arrow-left"
    onClick={[Function]}
    onKeyPress={[Function]}
    style={
      Object {
        "backgroundColor": "#848591",
        "pointerEvents": "none",
      }
    }
    tabIndex="1"
  >
    <i
      className="fa fa-angle-left"
    />
  </div>
  <div
    className="question"
  >
    <div
      className="title"
    >
      Q1
    </div>
    <ul
      className="options"
    >
      <li
        className=""
        key="Opt1"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="1"
      >
        <span>
          Opt1
        </span>
      </li>
      <li
        className=""
        key="Opt2"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="1"
      >
        <span>
          Opt2
        </span>
      </li>
    </ul>
  </div>
  <div
    className="arrow arrow-right"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="1"
  >
    <i
      className="fa fa-angle-right"
    />
  </div>
</div>
`;
